src/ui/common/src/components/pages/workflow/id/hook.ts

```diff
@@ -20,11 +28,58 @@ export type useWorkflowOutputs = {
   workflowDagResultWithLoadingStatus: WorkflowDagResultWithLoadingStatus;
 };
 
+// useWorkflowIds ensures we use the URL parameter as ground-truth for fetching
+// workflow, dag, and result IDs. It includes additional hooks to ensure
+// redux states are in-sync.
+// This hook should be used for all pages that need to access a single DAG (or DAG result)
+// data.
+export function useWorkflowIds(apiKey: string): useWorkflowIdsOutputs {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const {
+    id: wfIdParam,
+    dagId: dagIdParam,
+    dagResultId: dagResultIdParam,
+  } = useParams();
+
+  const { data: dagResults } = useDagResultsGetQuery(
+    { apiKey, workflowId: wfIdParam },
+    { skip: !wfIdParam }
+  );
+
+  // Select the first availale dag result if ID is not provided.
```

Comment:
Is the first available result the first or last run? Or is it a random run?

Reply:
yes, we need to supply the order by parameters